fraud/service.go

```diff
@@ -0,0 +1,189 @@
+package fraud
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"sync"
+
+	"github.com/celestiaorg/celestia-node/header"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+// headerFetcher aliases a function that is used to fetch ExtendedHeader from store
+type headerFetcher func(context.Context, uint64) (*header.ExtendedHeader, error)
+
+// topics allows to operate with pubsub connection and pubsub topics
+type topics struct {
```

Comment:
nit: move to another file?

Reply:
Yes, I thought about that. Will extract into `topics.go`